lib/testing/TestDataGenMissingFrames.hpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * @file
+ * @brief Config-driven insertion of missing frames
+ */
+
+#ifndef TEST_DATA_GEN_MISSING_FRAMES_HPP
+#define TEST_DATA_GEN_MISSING_FRAMES_HPP
+
+#include "Stage.hpp"
+#include "buffer.h"
+
+/**
+ * @class TestDataGenMissingFrames
+ * @brief Config-driven dropping of frames
+ *
+ * This stage can be interposed between two buffers to drop frames specified in the configuration.
+ *
+ * @par Buffers
+ * @buffer in_buf Input kotekan buffer, to be (selectively) copied over to out_buf
+ *     @buffer_format Array of @c any
+ *     @buffer_metadata none, or a class derived from @c kotekanMetadata
+ * @buffer out_buf Output kotekan buffer, with config-specified frames from in_buf missing
+ *     @buffer_format Array of @c datatype
+ *     @buffer_metadata none, or a class derived from @c kotekanMetadata
+ *
+ * @conf   missing_frames @c Vector of UInt32 (Default: empty). Frames to drop.
+ * @conf   drop_frame_chance @c Double (Default: 0). Chance of dropping a frame if not in the @c
+ *missing_frames list.
+ *
+ * @author James Willis, Davor Cubranic
+ **/
+
+class TestDataGenMissingFrames : public kotekan::Stage {
```

Comment:
The name of this should likely be changed to something like `TestMissingFrames` or `TestDropFrames`.   It's not really generating data.

Reply:
Renamed to `TestDropFrames`